### ISSUE TITLE
Zod 3.12.0 doesn't bind the scope anymore

### ISSUE THREAD

**Issue by danielepolencic** [colinhacks/zod#986]
This code used to work prior to 3.12.0, but it doesn't now:

```js
const z = require("zod");

const run = z.object({ a: z.number() }).safeParse;

console.log(run({ a: 1 }));
```

```bash
$ node run.js
/user/node_modules/zod/lib/types.js:184
            schemaErrorMap: this._def.errorMap,
                                 ^

TypeError: Cannot read property '_def' of undefined
    at ZodType.safeParse (/user/node_modules/zod/lib/types.js:184:34)
    at Object.<anonymous> (/user/test.js:5:13)
```

**Comment by JacobWeisenburger** [comment]
it looks like this was broken in 3.9.0, so the last version to support this was 3.8.2.

Here is the commit that likely caused the problem: https://github.com/colinhacks/zod/commit/d47eadc014b1e488576e303f3ae9241f6b573a04

Related to:
- https://github.com/colinhacks/zod/issues/808
- https://github.com/colinhacks/zod/issues/937

**Comment by JacobWeisenburger** [comment]
It looks like this was broken in 3.9.0, so the last version to support this was 3.8.2.

---
Can I ask why you are trying to assign `safeParse` to another variable?

Would this work:
```ts
const schema = z.object( { a: z.number() } )
const run = schema.safeParse.bind( schema )
console.log( run( { a: 1 } ) )
```
or maybe this:
```ts
const schema = z.object( { a: z.number() } )
console.log( schema.safeParse( { a: 1 } ) )
```

---
https://github.com/colinhacks/zod/commit/d47eadc014b1e488576e303f3ae9241f6b573a04 in `src/types.ts` likely caused this issue.

Related to:
- https://github.com/colinhacks/zod/issues/808
- https://github.com/colinhacks/zod/issues/937

**Comment by colinhacks** [comment]
Fixed in 3.12.1 - sorry about that.

**Comment by danielepolencic** [comment]
No worries, thanks for the quick fix!
